lib/src/functions/map.dart

```diff
@@ -57,10 +49,26 @@ final _set = BuiltInCallable.overloadedFunction("set", {
   },
 });
 
-final _merge = _function("merge", r"$map1, $map2", (arguments) {
-  var map1 = arguments[0].assertMap("map1");
-  var map2 = arguments[1].assertMap("map2");
-  return SassMap({...map1.contents, ...map2.contents});
+final _merge = BuiltInCallable.overloadedFunction("merge", {
+  r"$map1, $map2": (arguments) {
+    var map1 = arguments[0].assertMap("map1");
+    var map2 = arguments[1].assertMap("map2");
+    return SassMap({...map1.contents, ...map2.contents});
+  },
+  r"$map1, $args...": (arguments) {
+    var map1 = arguments[0].assertMap("map1");
+    var args = arguments[1].asList;
+    if (args.isEmpty) {
+      throw SassScriptException("Expected \$args to contain a key.");
+    } else if (args.length == 1) {
+      throw SassScriptException("Expected \$args to contain a map.");
+    }
+    var map2 = args.last.assertMap("map2");
+    var keys = args.sublist(0, args.length - 1);
+    var nestedMap = _getImpl(map1, keys).tryMap() ?? const SassMap.empty();
```

Comment:
This currently traverses the maps twice, once for `_getImpl()` and again for `_setImpl()`. If you changed `_setImpl()` to something like `SassMap _modify(SassMap map, List<Value> keys, SassValue modify(Value old))` then you could handle the traversal once in a more general way.

Reply:
Good call, thanks